faunadb-java/src/main/java/com/faunadb/client/FaunaClient.java

```diff
@@ -455,6 +456,9 @@ private void handleQueryErrors(int statusCode, String body) {
           if (ex instanceof ConnectException || ex instanceof TimeoutException) {
               throw new UnavailableException(ex.getMessage(), ex);
           }
+          if (ex instanceof CompletionException && ex.getMessage().contains("too many concurrent streams")) {
```

Comment:
Should we maybe check the cause of the `CompletionException` here as well? Something like:

```java
if (ex instanceof CompletionException && ex.getCause() instanceof IOException && ex.getMessage().contains("too many concurrent streams"))
```